packages/block-editor/src/components/inspector-controls/slot.js

```diff
@@ -31,7 +32,12 @@ export default function InspectorControlsSlot( {
 	if ( label ) {
 		return (
 			<BlockSupportToolsPanel group={ group } label={ label }>
-				<Slot { ...props } bubblesVirtually={ bubblesVirtually } />
+				<BlockSupportSlotContainer
```

Comment:
Does `BlockSupportSlotContainer` exist only to collocate the context hook with the Slot?

Reply:
At this point in time that's correct. I didn't find a better option given the tools panel context isn't created until the `ToolsPanel` is within the `BlockSupportToolsPanel` component.

With the current approach, there is also an issue with maintaining the order of rendered items in the panel when they are injected from multiple places. I'm definitely open to suggestions on improving this.

Another option I was planning to explore but expect some issues with was, to see if I could set up ref forwarding for the `BlockSupportToolsPanel` and pass it through the `as` prop to the virtual bubbling Slot. The idea being that the `BlockSupportToolsPanel` then passes the slot's ref on to a wrapping div and then creates the `ToolsPanel` with the fills inside that. The goal being the slot no longer introduces a `div` inside the `ToolsPanel` and there would no longer be a need to manually pass the `ToolsPanelContext` through the slot.

As I mentioned, I'm probably missing something with that idea. Let me know if there are any fatal flaws with it.

Reply:
I haven't had much luck trying to find a working solution to avoid this or the fact the virtual bubbling slot introduces a div when it would be preferential to avoid that inside the `ToolsPanel`.

One problem I ran into was that the [virtual bubbling slots](https://github.com/WordPress/gutenberg/blob/5cca2e1a8021679aaf2fc713c1a2c1cce168c374/packages/components/src/slot-fill/bubbles-virtually/slot.js) do not appear to accept a function as a child as per the [normal slot component](https://github.com/WordPress/gutenberg/blob/5cca2e1a8021679aaf2fc713c1a2c1cce168c374/packages/components/src/slot-fill/slot.js#L85). 

> Slot also accepts optional children function prop, which takes fills as a param. It allows to perform additional processing and wrap fills conditionally.

The [docs do not appear to make any distinction here](https://github.com/WordPress/gutenberg/blob/086b77ed409a70a6c6a6e74dee704851eff812f2/packages/components/src/slot-fill/README.md). @gziolo or @youknowriad can you confirm if this is an issue with the docs or whether the [virtual bubbling slot component](https://github.com/WordPress/gutenberg/blob/5cca2e1a8021679aaf2fc713c1a2c1cce168c374/packages/components/src/slot-fill/bubbles-virtually/slot.js) should also offer that?

While exploring this, it looked like it might be possible to leverage the `fills` retrieved within the inspector controls slot and do something with them from there but that felt like the wrong path to go down. As things stand now, this PR gets the block support tools panel and slots functional. We can iterate on this in follow up PRs.